### PR TITLE
[Dataset] Movielens

### PR DESCRIPTION
**Pull request by classicsong**
## Description
Movielens dataset loader

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/movielens.py [hunk 1]
```diff
@@ -0,0 +1,1018 @@
...
+        info_path = os.path.join(self.save_path,
+                                 self.save_name + '.pkl')
+        if os.path.exists(graph_path) and \
+            os.path.exists(info_path):
+            return True
+
+        return False
+
+    def save(self):
+        """save the graph list and the labels"""
```
Comment (hetong007) [inline python/dgl/data/movielens.py:235]: Duplicated docstring? Besides, `load` doesn't have one.

File: python/dgl/data/movielens.py [hunk 2]
```diff
@@ -0,0 +1,1018 @@
...
+
+    def _generate_dec_graph(self, rating_pairs):
+        ones = np.ones_like(rating_pairs[0])
+        user_movie_ratings_coo = sp.coo_matrix(
+            (ones, rating_pairs),
+            shape=(self.num_user, self.num_movie), dtype=np.float32)
+        return bipartite(user_movie_ratings_coo, 'user', 'rate', 'movie')
+
+    def _drop_unseen_nodes(self, orign_info, cmp_col_name, reserved_ids_set, label):
+        # print("  -----------------")
```
Comment (hetong007) [inline python/dgl/data/movielens.py:380]: Do we want to control these `print`s by `self.verbose` or just delete them?

File: python/dgl/data/movielens.py [hunk 3]
```diff
@@ -0,0 +1,1018 @@
...
+    The dataset stores MovieLens ratings in two types 
+    of graphs. The encoder graph contains rating value 
+    information in the form of edge types. The decoder 
+    graph stores plain user-movie pairs in the form of 
+    a bipartite graph with no rating information. All 
+    graphs have two types of nodes: "user" and "movie".
+    
+    Statistics
+    ----------
+    Nodes: xxx
```
Comment (hetong007) [inline python/dgl/data/movielens.py:688]: Will we update the stats?

**Comment by HuXiangkun** [comment]
Same as @hetong007 , others LGTM

**Pull request merged**